projects/ngrx.io/content/guide/store/testing.md

```diff
@@ -59,7 +61,126 @@ describe('Auth Guard', () => {
 });
 </code-example>
 
-### Using Store for Integration Testing
+### Example: Effect injecting state
+
+The mock store can simplify testing Effects that inject State using the RxJs `withLatestFrom` operator.  The example below shows the `addBookToCollectionSuccess$` effect displaying a different alert depending on the number of books in the collection state.
+
+<code-example header="collection.effects.ts">
+import { Injectable } from '@angular/core';
+import { Actions, Effect, ofType, createEffect } from '@ngrx/effects';
+import { map, withLatestFrom } from 'rxjs/operators';
+import { CollectionApiActions } from '@example-app/books/actions';
+import * as fromBooks from '@example-app/books/reducers';
+import { Store, select } from '@ngrx/store';
+
+@Injectable()
+export class CollectionEffects {
+  addBookToCollectionSuccess$ = createEffect(
+    () =>
+      this.actions$.pipe(
+        ofType(CollectionApiActions.addBookSuccess),
+        withLatestFrom(this.store.pipe(select(fromBooks.getCollectionBookIds))),
+        map(([action, bookCollection]) => {
```

Comment:
What about using `tap` here?

Reply:
The issue is the effect is expected to return an Action, and after using withLatestFrom, the type is [Action, string[]], so I need to map back to Action.

Reply:
Ah got it, you're right. Once #1689 is merged we could change the map 😄